src/litserve/cli.py

```diff
@@ -1,13 +1,22 @@
+import shutil
 import subprocess
 import sys
 
 from litserve.utils import is_package_installed
 
 
 def _ensure_lightning_installed():
-    if not is_package_installed("lightning_sdk"):
-        print("Lightning CLI not found. Installing...")
-        subprocess.check_call([sys.executable, "-m", "pip", "install", "-U", "lightning-sdk"])
+    """Ensure lightning-sdk is installed, attempting auto-installation if needed."""
+    if is_package_installed("lightning_sdk"):
+        return
+
+    print("Lightning CLI not found. Installing lightning-sdk...")
+    pip = ["uv", "pip"] if shutil.which("uv") else [sys.executable, "-m", "pip"]
```

Comment:
`shutil.which('uv')` only checks whether uv exists somewhere on the system PATH. If the user has uv installed globally but is running LitServe inside a `conda` environment or another package manager, this logic will still attempt to use uv, even if it isn’t part of the active environment."